plugin/core/registry.py

```diff
@@ -73,15 +73,13 @@ def client_from_session(session: Optional[Session]) -> Optional[Client]:
     return session.client if session else None
 
 
-def sessions_for_view(view: sublime.View, point: Optional[int] = None) -> Iterable[Session]:
-    return _sessions_for_view_and_window(view, view.window(), point)
+def sessions_for_view(view: sublime.View, capability: str = None, point: int = None) -> Iterable[Session]:
+    return (session for session in _sessions_for_view_and_window(view, view.window(), point)
+            if not capability or session.has_capability(capability))
 
 
-def session_for_view(view: sublime.View,
-                     capability: str,
-                     point: Optional[int] = None) -> Optional[Session]:
-    return next((session for session in sessions_for_view(view, point)
-                 if session.has_capability(capability)), None)
+def session_for_view(view: sublime.View, capability: str, point: int = None) -> Optional[Session]:
```

Comment:
So when we use default argument then we don't need `Optional` annotation? It works either way, not sure what's preferred though.

Reply:
Odd. It might be a mypy bug. It should definitely be `point: Optional[int] = None`, if only for communication to the reader.

Reply:
No, I think it works either way with mypy.